### DISCUSSION TITLE
SSE 프로젝트 주제를 정해봐요

### DISCUSSION THREAD

**Discussion by suna-ji** [EngineersRookieLab/realtime-stock-chart#2] [Ideas]
## 아이디어 목록
- **환율 변동 알림 시스템** 
https://github.com/EngineersRookieLab/realtime-spring-boot/discussions/2#discussioncomment-11164353
by @since1909 

- **재택 근무자 출퇴근 알림 시스템** 
https://github.com/EngineersRookieLab/realtime-spring-boot/discussions/2#discussioncomment-11164378
by @since1909 

- **주식 시세 실시간 알림 시스템**
https://github.com/EngineersRookieLab/realtime-spring-boot/discussions/2#discussioncomment-11166707
by @yuiseo 

- **출퇴근 경로 기반 CCTV 날씨 알림 시스템**
https://github.com/EngineersRookieLab/realtime-spring-boot/discussions/2#discussioncomment-11166805
by @yuiseo 

## 고려할 점
- 외부 API 무료 사용 가능한 범위 내에서 주제 선정
- front/back 간 기능 구현이 명확하도록 주제 선정

## 관련 링크
[UI 툴 - galileo ai](https://www.usegalileo.ai/explore)
[UI 툴 - creatie ai](https://creatie.ai/)

**Comment by suna-ji** [comment]
데이터 변동성이 큰 도메인을 선택하면 더 유의미한 프로젝트가 될 것 같아요!

**Comment by since1909** [comment]
### 환율 변동 시스템 알림 서비스

_제시한 아이디어 대해 좀 더 구체적으로 기술하겠습니다._

1. 원/달러, 원/엔 환율 정보를 제공
2. 사용자가 지정한 금액보다 올라가거나, 내려가면 알림 전송

---
- [openAPI 연동 가능](https://www.data.go.kr/data/3068846/openapi.do)
- 환율에 영향을 받는 다른 서비스 등과 연동하여 확장할 가능성 있음

**Comment by since1909** [comment]
### 재택 근무 출퇴근 알림 서비스

1. 출근/퇴근 시 사용에게 알림 전송
2. 휴가 전 근태 알림 전송

---
- 데이터의 변동성이 그치 않음
- 서비스 확장하기에는 제한적인 아이디어

**Comment by yuiseo** [comment]
## 주식 시세 실시간 알림 시스템
- 사용자가 지정한 주식의 가격 변동 시 실시간 알림.
- 외부 API: [Alpha Vantage](https://www.alphavantage.co/)나 [IEX Cloud](https://iexcloud.io/)의 무료 주식 시세 API.
---
- 백엔드 역할: 주기적으로 주식 시세를 확인하고 설정한 변동 범위에 도달하면 SSE로 알림을 전송.
- 프론트엔드 역할: 사용자가 원하는 주식을 추가/삭제할 수 있는 기능을 제공하며 실시간 시세 변동을 UI로 표현.

**Comment by yuiseo** [comment]
# 출퇴근 경로 기반 CCTV 날씨 알림 시스템

## 프로젝트 개요
- **목적**: 사용자가 출퇴근 시 지정한 위치의 CCTV 정보를 활용해 실시간 날씨 상황을 확인하고, 출퇴근 시간에 맞춰 사용자에게 알림 제공.
- **사용 API**: [공공 데이터포털 CCTV 날씨 정보 API](https://www.data.go.kr/data/15057966/openapi.do).
- **기능**: 특정 위치와 시간대를 설정해두면, 출퇴근 시간 전후에만 해당 위치의 날씨 정보를 실시간으로 받아 알림 제공.

## 주요 기능
1. **위치 설정 기능**  
   - 사용자가 자주 지나가는 출퇴근 경로상의 위치를 추가 (예: 지하철역, 주요 건물, 버스 정류장).
  
2. **출퇴근 시간 설정**  
   - 출근과 퇴근 시간을 설정하면, 설정한 시간 전후에만 알림을 제공하여 불필요한 알림을 줄임.

3. **CCTV 날씨 정보 연동**  
   - CCTV API를 통해 설정된 위치의 실시간 날씨 정보를 주기적으로 조회해 날씨 상태(강수, 온도, 가시거리 등)를 확인.

4. **알림 조건 설정**  
   - 특정 조건(예: 비/눈 예보, 기온 하락, 짙은 안개 등)을 사용자 설정에 따라 필터링해 조건에 맞는 경우만 알림 전송.

## 구현 방법

### 백엔드
- **위치 및 시간 기반 조회**  
  - 사용자 설정에 맞춰 출퇴근 시간 전후 특정 간격(예: 10분)에 해당 위치의 CCTV 날씨 정보를 API에서 조회.
  
- **조건 기반 필터링 및 알림 전송**  
  - 조회된 정보가 사용자가 설정한 조건(강수량, 온도, 시정 거리 등)을 충족하면 SSE 또는 WebSocket을 통해 프론트엔드로 알림 전송.

### 프론트엔드
- **위치 및 시간 설정 UI**  
  - 사용자가 위치와 출퇴근 시간, 알림 조건을 설정할 수 있는 인터페이스 제공.
  
- **실시간 알림 표시**  
  - CCTV 날씨 정보에 기반해 시각적으로 알림을 표시하고, 설정 조건에 맞는 알림만 보여줌.

## 추가 아이디어
- **위치 기반 알림 최적화**  
  - 사용자가 설정한 위치에 도착했을 때만 알림을 받도록 위치 기반 기능(GPS 등)을 추가하여 불필요한 알림을 최소화.

- **유사 정보 병합**  
  - 같은 지역의 여러 조건을 설정한 경우, 하나의 알림으로 병합하여 제공.

- **날씨 변화 예측**  
  - CCTV 날씨 정보와 함께 간단한 날씨 변화 예측 기능을 추가하여 비나 눈이 예상될 경우 대비할 수 있도록 알림.

---
확인이 필요한 사안: 모든 cctv 정보에서 날씨 정보를 가져오는지, cctv 번호를 어떻게 알아낼 수 있을지

**Comment by suna-ji** [comment]
## ⭐️결정 사항⭐️
- 프로젝트 주제
  -  주식 시세 실시간 알림 시스템
- API
  - [한투증 개발자 센터](https://apiportal.koreainvestment.com/about) 
  - [한투증 API Git Repo](https://github.com/koreainvestment/open-trading-api) 
- 목표
  - 순위, 종목명, 현재가, 전일 대비 보여주는 메인 페이지 
  - 세부 페이지 개발
